polaris-react/src/components/Box/Box.tsx

```diff
@@ -73,53 +66,60 @@ interface Spacing {
   top: SpacingTokenScale;
 }
 
+type Element = 'div' | 'span';
+
 export interface BoxProps {
-  as?: 'div' | 'span';
-  /** Background color of the Box */
+  /** HTML Element type */
+  as?: Element;
+  /** Adjust background color */
```

Comment:
I think 
```suggestion
  /** Background color */
```
Is fine for descriptions like these and we can drop `Adjust`

`Border style` would be fine for the below. Let's keep things simple